### PR TITLE
Watch installation manifest file changes

### LINKED ISSUE #7: watch for installation manifest file changes
Current implementation requires the user to reload vscode so newly created/added packages are properly recognized by the extension

### PR DESCRIPTION
**Pull request by g-arjones**
Closes #7

### CODE REVIEW

File: src/test/packages.test.ts [hunk 1]
```diff
@@ -134,15 +134,6 @@ describe("PackageFactory", function () {
                 let aPackage = await subject.createPackage(path, s.context);
                 assert.equal((aPackage as packages.RockPackage).ws, ws);
             })
-            it("attempts to regenerate the manifest if the package is not present in it", async function() {
```
Comment (doudou) [inline src/test/packages.test.ts]: I don't think you can get rid of this. All of this is asynchronous, one might go into this codepath before the other file-watching code path kicked in.

Comment (g-arjones) [reply src/test/packages.test.ts]: Problem here is that it creates a loop... The FileWatcher makes PackageFactory re-resolve the packages when installation-manifest changes and PackageFactory calls `autoproj envsh`, which updates de installation-manifest again. I think a better aproach to keep the package info up-to-date is #29

Comment (g-arjones) [reply src/test/packages.test.ts]: We run envsh when a new workspace is added (and on initialization) and watch config.yml/manifest. As soon as they change, we run envsh again

**Comment by doudou** [comment]
Cool ... just one comment (not sure how the async part would work out).

Also, idea for a longer term solution to that and the exec/which performance: https://github.com/rock-core/autoproj/issues/164

File: src/test/packages.test.ts (hunk 1 above)
Comment (doudou) [reply src/test/packages.test.ts]: The issue here is that other things than the manifest may change the list of packages (I'm especially thinking of `manifest.xml`).

This is getting too complicated. I think it is logic that should be in autoproj (https://github.com/rock-core/autoproj/issues/164)

Comment (g-arjones) [reply src/test/packages.test.ts]: > (I'm especially thinking of manifest.xml)

Yeah, that's probably the most common use-case. What do you suggest in the short term?

Comment (doudou) [reply src/test/packages.test.ts]: > What do you suggest in the short term?

I'd rather keep the status quo. We can't safely auto-reload installation-manifest because of the looping issue, and we don't really know when envsh must be run. That's too many corner cases.

Comment (g-arjones) [reply src/test/packages.test.ts]: Alrighty... 👍

**Comment by g-arjones** [comment]
@doudou I have rebased it... Could you please have a look again?
